queue_services/entity-filer/src/entity_filer/resources/worker.py

```diff
@@ -0,0 +1,143 @@
+import asyncio
+import re
+from dataclasses import dataclass
+from http import HTTPStatus
+from typing import Optional
+
+from flask import Blueprint, current_app, request
+from gcp_queue import SimpleCloudEvent
+from structured_logging import StructuredLogging
+
+from entity_filer.services import gcp_queue, verify_gcp_jwt
+from entity_filer.worker import process_filing
+
+
+bp = Blueprint('worker', __name__)
+logger = StructuredLogging.get_logger()
+loop = asyncio.get_event_loop()
+
+
+@bp.route('/', methods=('POST',))
+def worker():
+    """Process the incoming cloud event.
+
+    Flow
+    --------
+    1. Get cloud event
+    2. Get filing message information
+    3. Process filing
+
+    Decisions on returning a 2xx or failing value to
+    the Queue should be noted here:
+    - Empty or garbaled messages are knocked off the Q
+    - Filings that cannot be processed are knocked off the Q
+    """
+    if not request.data:
+        logger.debug('No incoming raw msg.')
+        return {}, HTTPStatus.OK
+
+    if msg := verify_gcp_jwt(request):
+        logger.info(msg)
+        return {}, HTTPStatus.FORBIDDEN
+
+    logger.info(f'Incoming raw msg: {str(request.data)}')
+
+    # 1. Get cloud event
+    # ##
+    if not (ce := gcp_queue.get_simple_cloud_event(request,
+                                                   wrapped=True)) \
+            and not isinstance(ce, SimpleCloudEvent):
+        #
+        # Decision here is to return a 200,
+        # so the event is removed from the Queue
+        logger.debug(f'ignoring message, raw payload: {str(ce)}')
+        return {}, HTTPStatus.OK
+    logger.info(f'received ce: {str(ce)}')
+
+    # 2. Get filing_message information
+    # ##
+    if not (filing_message := get_filing_message(ce)):
+        # no filing_message info, error off Q
+        logger.debug(f'no filing_message info in: {ce}')
+        return {'error': 'no filing info in cloud event'}, HTTPStatus.BAD_REQUEST
+    logger.info(f'Incoming filing_message: {filing_message}')
+
+    # 3. Process Filing
+    # ##
+    try:
+        loop.run_until_complete(process_filing(filing_message, current_app))
+    except Exception as err:  # pylint: disable=broad-exception-caught
+        logger.error(f'Error processing filing {filing_message}: {err}')
+        return {'error': f'Unable to process filing: {filing_message}'}, HTTPStatus.BAD_REQUEST
+
+    logger.info(f'completed ce: {str(ce)}')
+    return {}, HTTPStatus.OK
+
+
+@dataclass
+class FilingMessage:
+    """FilingMessage object from the cloud event."""
+
+    id: Optional[str] = None
+    status_code: Optional[str] = None
+    filing_identifier: Optional[str] = None
+    corp_type_code: Optional[str] = None
+
+
+def get_filing_message(ce: SimpleCloudEvent):
+    """Return a FilingMessage if enclosed in the cloud event."""
+    if (
+        (ce.type == 'filingMessage')
+        and isinstance(ce.data, dict)
+        and (filing_message := ce.data.get('filingMessage', {}))
+    ):
```

Comment:
Just a quick question on the message format communicated between business-pay and filer.

When I updated business-pay, I keep the old format. As a result, the message will be something like
```json
{
    "specversion": "1.0",
    "datacontenttype": "application/json",
    "source": "business-pay",
    "subject": <business-filer-topic>,
    "type": "filing-message",
    "data": {
        "filing": {
            "id": 123
        }
    }
}
```
Which format should we use? @argush3

Reply:
@chenhongjing I think business-pay should be updated to follow the format Leo is using here.